### PR TITLE
Split giant chemistry files into oldchem folder

### PR DESCRIPTION
**Pull request by tigercat2000**
This commit does the following:
 - Removes relative pathing from:
  - Chemistry-Holder.dm
  - grenade_launcher.dm
  - reagent_dispenser.dm
 - Splits Chemistry-Reagents.dm into oldchem/reagents/
 - Splits Chemistry-Recipies.dm into oldchem/chemical_reaction/
  - These are still relative-pathed, but split into files, so removing the relative pathing should just be as easy as finding someone with the energy to do it.

Another couple of huge relative-pathed files resolved.

@Fox-McCloud @Iamgoofball Would really love a review to make sure I didn't fuck anything up, and you two seem to know chem code.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
If you're eliminating relative pathing...why are a lot of the reaction still relatively pathed?

**Comment by tigercat2000** [comment]
Because it took about an hour to do the reagents, which were a smaller file. Splitting it is still better than one giant file.

File: code/modules/reagents/oldchem/reagents/reagents_drugs.dm [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+/datum/reagent/lithium/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(M.canmove && !M.restrained() && istype(M.loc, /turf/space))
+		step(M, pick(cardinal))
+	if(prob(5)) M.emote(pick("twitch","drool","moan"))
+	..()
+	return
+
+
+/datum/reagent/hippies_delight
```
Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/reagents/reagents_drugs.dm]: this is a drink, not a drug.

File: code/modules/reagents/oldchem/chemical_reaction/chemical_reaction_misc.dm [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+				for(var/mob/M in viewers(get_turf(holder.my_atom), null))
+					M.show_message( "<span class='danger'>The blood writhes and wriggles and sizzles away from the container!</span>", 1, "<span class='warning'>You hear bubbling and sizzling.</span>", 2)
+			else
+				for(var/mob/M in viewers(get_turf(holder.my_atom), null))
+					M.show_message( "<span class ='notice'>The blood seems to break apart in the fuel.</span>", 1)
+			holder.del_reagent("blood")
+			return
+*/
+
+		plastication
```
Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/chemical_reaction/chemical_reaction_misc.dm]: indented twice?

**Comment by Fox-McCloud** [comment]
I'll look this over later. Sadly Github is shit about properly showing changes when there's large amounts changes so...yeah.

File: code/modules/reagents/oldchem/reagents/reagents_drugs.dm (hunk 1 above)
Comment (tigercat2000) [inline code/modules/reagents/oldchem/reagents/reagents_drugs.dm]: Wasn't actually part of the drink subtype, I could move it to the drink file, I guess.

Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/reagents/reagents_drugs.dm]: That's...weird. I'd definitely recommend making it a drink sub-type if you don't mind.

File: code/modules/reagents/Chemistry-Holder.dm [hunk 3]
```diff
@@ -5,621 +5,618 @@ var/const/INGEST = 2
...
+	// the same donor.
+	//
+	// Technically we should probably copy all data lists, but
+	// that could possibly eat up a lot of memory needlessly
+	// if most data lists are read-only.
+	if (trans_data["virus2"])
+		var/list/v = trans_data["virus2"]
+		trans_data["virus2"] = v.Copy()
+
+	return trans_data
```
Comment (Fox-McCloud) [inline code/modules/reagents/Chemistry-Holder.dm]: This one looks good.

File: code/modules/reagents/grenade_launcher.dm [hunk 4]
```diff
@@ -17,57 +17,57 @@
...
+	F.loc = user.loc
+	F.throw_at(target, 30, 2, user)
+	message_admins("[key_name_admin(user)] fired a [ammo_name] ([F.name]) from a launcher ([name]).")
+	log_game("[key_name_admin(user)] used a [ammo_name] ([name]).")
+	F.active = 1
+	F.icon_state = initial(icon_state) + "_active"
+	playsound(user.loc, 'sound/weapons/armbomb.ogg', 75, 1, -3)
+	spawn(15)
+		F.prime()
+
```
Comment (Fox-McCloud) [inline code/modules/reagents/grenade_launcher.dm]: This one looks good.

File: code/modules/reagents/oldchem/chemical_reaction/_chemical_reaction_base.dm [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	var/secondary = 0 // set to nonzero if secondary reaction
+	var/list/secondary_results = list()		//additional reagents produced by the reaction
+	var/min_temp = 0		//Minimum temperature required for the reaction to occur (heat to/above this). min_temp = 0 means no requirement
+	var/max_temp = 9999		//Maximum temperature allowed for the reaction to occur (cool to/below this).
+	var/mix_message = "The solution begins to bubble."
+	var/mix_sound = 'sound/effects/bubbles.ogg'
+	var/no_message = 0
+
+/datum/chemical_reaction/proc/on_reaction(var/datum/reagents/holder, var/created_volume)
+	return
```
Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/chemical_reaction/_chemical_reaction_base.dm]: this one looks good.

File: code/modules/reagents/oldchem/reagents/__oldchem_defines.dm [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#define SOLID 1
+#define LIQUID 2
+#define GAS 3
+#define FOOD_METABOLISM 0.4
+#define REM REAGENTS_EFFECT_MULTIPLIER
```
Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/reagents/__oldchem_defines.dm]: this one looks good

File: code/modules/reagents/oldchem/reagents/_reagent_base.dm [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+/datum/reagent/proc/on_move(var/mob/M)
+	return
+
+/datum/reagent/proc/on_update(var/atom/A)
+	return
+
+/datum/reagent/Destroy()
+	holder = null
+	return ..()
```
Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/reagents/_reagent_base.dm]: this one looks good.

File: code/modules/reagents/oldchem/reagents/drink/reagents_alcohol.dm [hunk 8]
```diff
@@ -0,0 +1,748 @@
...
+	data++
+	M.dizziness +=10
+	if(data >= 55 && data <115)
+		if (!M.stuttering) M.stuttering = 1
+		M.stuttering += 10
+	else if(data >= 115 && prob(33))
+		M.confused = max(M.confused+15,15)
+	..()
+
+	return
```
Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/reagents/drink/reagents_alcohol.dm]: this one looks good.

File: code/modules/reagents/oldchem/reagents/drink/reagents_drink_base.dm [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+	if (adj_dizzy) M.dizziness = max(0,M.dizziness + adj_dizzy)
+	if (adj_drowsy)	M.drowsyness = max(0,M.drowsyness + adj_drowsy)
+	if (adj_sleepy) M.sleeping = max(0,M.sleeping + adj_sleepy)
+	if (adj_temp)
+		if (M.bodytemperature < 310)//310 is the normal bodytemp. 310.055
+			M.bodytemperature = min(310, M.bodytemperature + (25 * TEMPERATURE_DAMAGE_COEFFICIENT))
+	// Drinks should be used up faster than other reagents.
+	holder.remove_reagent(src.id, FOOD_METABOLISM)
+	..()
+	return
```
Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/reagents/drink/reagents_drink_base.dm]: this one looks good.

File: code/modules/reagents/oldchem/reagents/drink/reagents_drink.dm [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+	description = "A widely known, Mexican coffee-flavoured liqueur. In production since 1936!"
+	color = "#664300" // rgb: 102, 67, 0
+	adj_dizzy = -5
+	adj_drowsy = -3
+	adj_sleepy = -2
+
+/datum/reagent/drink/kahlua/on_mob_life(var/mob/living/M as mob)
+	..()
+	M.Jitter(5)
+	return
```
Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/reagents/drink/reagents_drink.dm]: this one looks good.

File: code/modules/reagents/oldchem/reagents/drink/reagents_drink_cold.dm [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+/datum/reagent/drink/cold/rewriter
+	name = "Rewriter"
+	description = "The secert of the sanctuary of the Libarian..."
+	id = "rewriter"
+	color = "#485000" // rgb:72, 080, 0
+
+/datum/reagent/drink/cold/rewriter/on_mob_life(var/mob/living/M as mob)
+	..()
+	M.Jitter(5)
+	return
```
Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/reagents/drink/reagents_drink_cold.dm]: looks good.

File: code/modules/reagents/oldchem/reagents/reagents_admin.dm [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+				var/datum/disease2/disease/V = C.virus2[ID]
+				C.antibodies |= V.antigen
+	..()
+	return
+
+
+/datum/reagent/adminordrazine/nanites
+	name = "Nanites"
+	id = "nanites"
+	description = "Nanomachines that aid in rapid cellular regeneration."
```
Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/reagents/reagents_admin.dm]: looks good.

File: code/modules/reagents/reagent_dispenser.dm [hunk 13]
```diff
@@ -241,6 +248,7 @@
 	anchored = 1
 	density = 0
 	amount_per_transfer_from_this = 250
-	New()
-		..()
-		reagents.add_reagent("cleaner",5000)
+
+/obj/structure/reagent_dispensers/spacecleanertank/New()
+	..()
+	reagents.add_reagent("cleaner",5000)
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_dispenser.dm]: this one looks good.

**Comment by Fox-McCloud** [comment]
This looks good to me; I believe all the procs are correct and the indentation is good. I'm not going to line by line every single reagent, but it appears you got them all---any that you didn't get are sure to show up in the runtime logs or player issue reports (if you did miss any).

**Comment by Iamgoofball** [comment]
looks about right

**Comment by TheDZD** [comment]
Merge conflicts.

**Comment by tigercat2000** [comment]
Less merge conflicts.

**Pull request merged**